src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -130,14 +130,30 @@ internal virtual string GetAssemblyFileVersion()
         {
             if (_clientDirectory != null)
             {
-                var name = $"{Type.GetTypeInfo().Assembly.GetName().Name}.dll";
+                Assembly assembly = Type.GetTypeInfo().Assembly;
+                var name = $"{assembly.GetName().Name}.dll";
                 var filePath = Path.Combine(_clientDirectory, name);
-                return FileVersionInfo.GetVersionInfo(filePath).FileVersion;
+                var fileVersionInfo = FileVersionInfo.GetVersionInfo(filePath);
+                string hash = ExtractShortCommitHash(assembly);
+
+                return $"{fileVersionInfo.FileVersion} ({hash})";
             }
 
             return "";
         }
 
+        private static string ExtractShortCommitHash(Assembly assembly)
+        {
+            // leave "<developer build>" alone, but truncate SHA to 8 characters
+            string hash = assembly.GetCustomAttribute<CommitHashAttribute>()?._hash;
+            if (hash != null && hash.Length >= 8 && hash[0] != '<')
+            {
+                hash = hash.Substring(0, 8);
+            }
```

Comment:
Expose an internal `ExtractShortCommitHash` method that takes an optional string (or optional `CommitHashAttribute`) and test these cases.